### PR TITLE
fix(discover): Allow sorting by user.display

### PR DESCRIPTION
**Pull request by Zylphrex**
Currently, ordering by `user.display` is parsed to `tags[user.display]`. This
will not error when there are no aggregates, but the resulting order is sorting
on empty strings. When there are aggregates, the ordering by `tags[user.display]`
results in a column that is not in any aggregate/group bys making for an invalid
query. This change will resolve `user.display`  to `user.display` for correct
ordering.

### CODE REVIEW

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 1]
```diff
@@ -1539,6 +1539,84 @@ def test_user_display_with_aggregates(self):
...
+            "statsPeriod": "24h",
+            "orderby": "-user.display",
+        }
+        response = self.do_request(query, features=features)
+        assert response.status_code == 200, response.content
+        data = response.data["data"]
+        assert len(data) == 2
+        result = [r["user.display"] for r in data]
+        # because we're ordering by `-user.display`, we expect the results in reverse sorted order
+        assert result == list(reversed(sorted(["catherine", "cathy@example.com"])))
```
Comment (wmak) [inline tests/snuba/api/endpoints/test_organization_events_v2.py:1579]: 🐎 
```suggestion
        assert result == ['cathy@example.com', 'catherine']
```

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 2]
```diff
@@ -1539,6 +1539,84 @@ def test_user_display_with_aggregates(self):
...
+            "statsPeriod": "24h",
+            "orderby": "user.display",
+        }
+        response = self.do_request(query, features=features)
+        assert response.status_code == 200, response.content
+        data = response.data["data"]
+        assert len(data) == 2
+        result = [r["user.display"] for r in data]
+        # because we're ordering by `user.display`, we expect the results in sorted order
+        assert result == list(sorted(["catherine", "cathy@example.com"]))
```
Comment (wmak) [inline tests/snuba/api/endpoints/test_organization_events_v2.py:1618]:
```suggestion
        assert result == ["catherine", "cathy@example.com"]
```

**Review by wmak — APPROVED** [review]
✅

**Review by evanh — APPROVED** [review]
Quick turnaround, nice.

**Review by k-fish — APPROVED** [review]
More tests than code? that's 🥇

**Pull request merged**